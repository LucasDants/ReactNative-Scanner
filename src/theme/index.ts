export default {
  colors: {
    red: {
      900: '#B83341',
      800: '#E03F50',
    },
    blue: '#0000FF',
    white: '#FFFFFF',
    gray: {
      100: '#DCDCDC',
      600: '#404040',
      900: '#030303',
    },
  },
};