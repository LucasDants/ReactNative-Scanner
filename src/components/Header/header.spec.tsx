import React from 'react'

import { render } from '@testing-library/react-native'

import { ThemeProvider } from 'styled-components/native'
import theme from '../../theme'

import { Header } from '.'

const Providers = ({ children }) => (
  <ThemeProvider theme={theme}>
    { children }
  </ThemeProvider>
)

describe('Header Component', () => {

  it("should match the snapshot test", () => {
    const { toJSON } = render(
      <Header title='Test' />, {wrapper: Providers}
    )

    const component = toJSON()
    expect(component).toMatchSnapshot()

  })

  it("should have the correct title", () => {
    const { getByText } = render(
      <Header title='Test' />, {wrapper: Providers}
    )

    expect(getByText('Test')).toBeTruthy()

  })

})