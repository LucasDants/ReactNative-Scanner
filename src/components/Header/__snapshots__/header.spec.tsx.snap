// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot test 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4290261825,
        4292886352,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#B83341",
        "borderBottomWidth": 1,
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 33,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 51,
        },
      ]
    }
  >
    Test
  </Text>
</ViewManagerAdapter_ExpoLinearGradient>
`;
